wp-includes/class-wp-service-workers.php

```diff
@@ -83,6 +148,55 @@ public function register( $handle, $src, $deps = array(), $scope = self::SCOPE_A
 		return parent::add( $handle, $src, $deps, false, compact( 'scope' ) );
 	}
 
+	/**
+	 * Register route and caching strategy.
+	 *
+	 * @todo Maybe route should be an array -- $routes -- instead, this way the method could also be used for precaching.
+	 * @param string $route Route.
+	 * @param int    $strategy Strategy, can be self::STRATEGY_STALE_WHILE_REVALIDATE (@todo Add others).
+	 * @param array  $args Array of args, can be cache_name, max_age, max_entries.
+	 */
+	public function register_route( $route, $strategy, $args ) {
+
+		// @todo Should we support custom callbacks? Maybe WP_Service_Worker::register() is sufficient already.
+		// @todo Support pre-caching.
+		// @todo Add other strategies.
+		// @todo Add validation.
+		// @todo Logic for detecting conflicts.
+		switch ( $strategy ) {
+			default:
+				$strategy = 'staleWhileRevalidate';
+				break;
+		}
+
+		$this->caching_strategies .= $this->register_caching_strategy_for_route( $route, $strategy, $args );
+
+	}
+
+	/**
+	 * Register caching strategy for route.
+	 *
+	 * @param string $route Route.
+	 * @param int    $strategy Strategy, can be self::STRATEGY_STALE_WHILE_REVALIDATE (@todo Add others).
+	 * @param array  $args Array of args, can be cache_name, max_age, max_entries.
+	 * @return string Script.
+	 */
+	protected function register_caching_strategy_for_route( $route, $strategy, $args ) {
+		$script = "wp.serviceWorker.addCachingStrategy( '" . $route . "', '" . $strategy . "'";
```

Comment:
Note that `wp_json_encode()` needs to be used here instead of manually including quotes. This will ensure that if a route contains quotes it won't break the JS, for example.